### ISSUE TITLE
Is there a way to forbid certain usages of mixins?

### ISSUE THREAD

**Issue by sandrooco** [stylelint/stylelint#7017]
### Describe the documentation issue

When using this: `@media screen and (max-width: $desktop) {` we will correctly get the error `Dont use "max-width". Always mobile first` from the rule `media-feature-name-disallowed-list`.

The problem: we have an internal library that simplifies using media queries. The same styling will be achieved like this:
`@include mq($to: $desktop) {` but obviously `media-feature-name-disallowed-list` doesn't understand that this is just a max-width media query.
Is there a way to ban the usage of the `mq` mixin with (and only with) `$to`? Unfortunately I can't change the library.

### What solution would you like to see?

A way that enables me to ban all usages of `mq($to: *)`.

**Comment by sandrooco** [comment]
(Of course we could rebuild it all with min-width media queries but it's a large project so we want to update it step by step.)

**Comment by ybiquitous** [comment]
@sandrooco Do you want a rule like `stylelint-scss`'s [`at-mixin-pattern`](https://github.com/stylelint-scss/stylelint-scss/blob/v5.0.1/src/rules/at-mixin-pattern/README.md)?

Stylelint's built-in rules support only the standard CSS syntax. So you may need to find a rule  from `stylelint-scss` or [write a custom plugin](https://stylelint.io/developer-guide/plugins) by using `postcss-scss`.

**Comment by sandrooco** [comment]
No it unfortunately does not. I only want to block usages with the `$to`. Thanks for the quick feedback!

**Comment by jeddy3** [comment]
>  I only want to block usages with the $to

You can [create a plugin](https://stylelint.io/developer-guide/plugins). The logic of the custom rule in it could be as simple as:

```js
root.walkAtRules("include", (atRule) => {

  if (!atRule.params.startsWith('mq($to')) return;

  utils.report({
    message: messages.rejected,
    node: atRule,
    result,
    ruleName,
  });
}
```

[AST for reference](https://astexplorer.net/#/gist/fcd4257738715162eeda2d23065cb488/2819f2ad78bed184a77f4a2cdd7d1bf795079a44).

**Comment by sandrooco** [comment]
@jeddy3 thanks for the input! I tried it and it worked.
Is there any documentation and/or example on how to write, build and deploy plugins with typescript? I was only getting it to work with js.

**Comment by jeddy3** [comment]
> Is there any documentation and/or example on how to write, build and deploy plugins with typescript? 

I don't know of any offhand. I'll open an issue as it's something that would be useful to plugin authors, especially as we publish our types.
